vtex/faststore
# Feat: Pages with `RenderSection` + `Breadcrumb` section CMS

## What's the purpose of this pull request?

You may notice that in some pages (e.g. PDP) the breadcrumb is not shown. That's because CMS team is investigating some behaviors and maybe the Breadcrumb section [is not present in CMS](https://storeframework.myvtex.com/admin/new-cms/faststore/pdp/edit/717bc97f-8f19-4744-a52b-7ccac43cc898/).

- [x] Adjusts the `Breadcrumb` section to be rendered using CMS data.
- [x] Moves `Breadcrumb` from `ProductDetails` to `PDP` page.
- [x] Adapts `Breadcrumb` in `PDP`, `PLP` and `Search` pages.
- [x] Prepare pages to be rendered using `RenderSection` with CMS data (CMS).
- [x] Adds Search content-type.
- [x] Promise all sequential requests from server side (static) pages (performance: build time).

## How it works?

It gets data from CMS and pass to `RenderSections` components, which in turn pass to `Breadcrumb` section. As we have this component in some pages, we also pass a `context` prop depending on the page:
- PLP - context = collection
- PDP - context = product
- Search - context = undefined.

## How to test it?

run `yarn dev` into the core package and check the breadcrumb in these pages.
you can also see the starter preview.

### Starters Deploy Preview

PR https://github.com/vtex-sites/starter.store/pull/54
Preview https://preview-54--starter.preview.vtex.app/
